discovery-provider/src/tasks/index.py

```diff
@@ -273,89 +297,101 @@ def index_blocks(self, db, blocks_list):
+                logger.info(
+                    f"index.py | playlist_state_update completed"
+                    f" playlist_state_changed={playlist_state_changed} for block={block_number}"
+                )
 
-            track_lexeme_state_changed = (user_state_changed or track_state_changed)
-            session.commit()
-            logger.info(f"index.py | session commmited to db for block=${block_number}")
-
-            if user_state_changed:
-                if user_ids:
-                    remove_cached_user_ids(redis, user_ids)
-            if user_replica_set_state_changed:
-                if replica_user_ids:
-                    remove_cached_user_ids(redis, replica_user_ids)
-            if track_lexeme_state_changed:
-                if track_ids:
-                    remove_cached_track_ids(redis, track_ids)
-            if playlist_state_changed:
-                if playlist_ids:
-                    remove_cached_playlist_ids(redis, playlist_ids)
-            logger.info(f"index.py | redis cache clean operations complete for block=${block_number}")
+                user_library_state_changed = user_library_state_update( # pylint: disable=W0612
+                    self, update_task, session, user_library_factory_txs, block_number, block_timestamp, block_hash
+                )
+                logger.info(
+                    f"index.py | user_library_state_update completed"
+                    f" user_library_state_changed={user_library_state_changed} for block={block_number}"
+                )
 
+                track_lexeme_state_changed = (user_state_changed or track_state_changed)
+                session.commit()
+                logger.info(f"index.py | session commmited to db for block=${block_number}")
+                if skip_tx_hash:
+                    clear_indexing_error(redis)
+                if user_state_changed:
+                    if user_ids:
+                        remove_cached_user_ids(redis, user_ids)
+                if user_replica_set_state_changed:
+                    if replica_user_ids:
+                        remove_cached_user_ids(redis, replica_user_ids)
+                if track_lexeme_state_changed:
+                    if track_ids:
+                        remove_cached_track_ids(redis, track_ids)
+                if playlist_state_changed:
+                    if playlist_ids:
+                        remove_cached_playlist_ids(redis, playlist_ids)
+                logger.info(f"index.py | redis cache clean operations complete for block=${block_number}")
+            except IndexingError as err:
+                logger.info(
+                    f"index.py | Error in the indexing task at"
+                    f" block={err.blocknumber} and hash={err.transactionhash}"
+                )
+                set_indexing_error(redis, err.blocknumber, err.blockhash, err.transactionhash, err.message)
+                confirm_indexing_transaction_error(
```

Comment:
confirm_indexing_transaction_error also conditionally calls set_transaction_error, in which case I'm not sure we'd want to also call set_transaction_error on the line directly above

Reply:
Well this function calls `set_indexing_error` w/ the last parameter `has_majority` - this field dictates if the indexing flow should try and skip the transaction. Let me know if this makes sense

Reply:
I was thinking about the line that increments the count for the indexing error. I think the way the code is currently written it's possible to set_indexing_error here on line 391, the confirm_indexing_transaction_error can also call set_indexing_error if has_majority is true and the count for the indexing error would be 2 instead of 1. Maybe this is intended behavior? but I didn't think so

Reply:
This is intended behavior